pandas/tests/reshape/merge/test_merge.py

```diff
@@ -1763,3 +1763,23 @@ def test_merge_equal_cat_dtypes2():
 
     # Categorical is unordered, so don't check ordering.
     tm.assert_frame_equal(result, expected, check_categorical=False)
+
+
+@pytest.mark.parametrize('merge_type', ['left_on', 'right_on'])
+def test_merge_correct_exception(merge_type):
+    # GH26824
+    df1 = DataFrame({
+        'A': [1, 2, 3, 4, 5, 6],
+        'B': ['P', 'Q', 'R', 'S', 'T', 'U']
+    })
+    df2 = DataFrame({
+        'A': [1, 2, 4, 5, 7, 8],
+        'C': ['L', 'M', 'N', 'O', 'P', 'Q']
+    })
+    msg = 'both left_on and right_on should be passed'
+    if merge_type == 'left_on':
```

Comment:
Rather than the if condition since you are parmatrizing this argument (which is nice btw) you can send it through as kwargs. So:

```python
kwargs = {merge_type: 'A'}
with pytest.raises(ValueError, match=msg):
    pd.merge(df1, df2, how='left', **kwargs)
```

Reply:
Makes sense, will do